### PR TITLE
Thheinen/efs accesspoints

### PR DESCRIPTION
**Pull request by mapk-amazon**
"Clone" of https://github.com/aws/aws-parallelcluster/pull/6359

### Description of changes
Add support to specify EFS access points per issue https://github.com/aws/aws-parallelcluster/issues/2337

### Tests
* Integration tests included in the `develop.yaml` as well as unit tests.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -321,6 +321,9 @@ class EfsSettingsSchema(BaseSchema):
     deletion_policy = fields.Str(
         validate=validate.OneOf(DELETION_POLICIES), metadata={"update_policy": UpdatePolicy.SUPPORTED}
     )
+    access_point_id = fields.Str(
+        validate=validate.Regexp(r"^fsap-[0-9a-z]{17}$"), metadata={"update_policy": UpdatePolicy.SUPPORTED}
+    )
```
Comment (dreambeyondorange) [inline cli/src/pcluster/schemas/cluster_schema.py]: Has this update support been tested?

File: cli/src/pcluster/validators/efs_validators.py [hunk 2]
```diff
@@ -18,10 +18,16 @@ class EfsMountOptionsValidator(Validator):
...
                 "EFS IAM authorization cannot be enabled when encryption in-transit is disabled. "
                 f"Please either disable IAM authorization or enable encryption in-transit for file system {name}",
                 FailureLevel.ERROR,
             )
+        if access_point_id and not name:
+            self._add_failure(
+                "An access point can only be specified when using an existing EFS file system. "
+                f"Please either remove the access point id {access_point_id} or provide the file system id for the access point",
+                FailureLevel.ERROR,
+            )
```
Comment (dreambeyondorange) [inline cli/src/pcluster/validators/efs_validators.py]: This probably needs to reference the `file_system_id` param

File: cli/tests/pcluster/validators/test_efs_validators.py [hunk 3]
```diff
@@ -42,8 +42,42 @@
         ),
     ],
 )
-def test_efs_mount_options_validator(encryption_in_transit, iam_authorization, expected_message):
+def test_efs_mount_options_validator(
```
Comment (dreambeyondorange) [inline cli/tests/pcluster/validators/test_efs_validators.py]: This test would need to be updated to reflect the `file_system_id` param above.

**Review by dreambeyondorange — CHANGES_REQUESTED** [review]
Sorry for the delay.  I think this PR looks great.  Thanks for the contribution.  I have a couple minor comments and once resolved I can approve!

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 1 above)
Comment (mapk-amazon) [reply cli/src/pcluster/schemas/cluster_schema.py]: Has not been tested. To be one the safe side I marked it as `unsupported` now.

File: cli/src/pcluster/validators/efs_validators.py (hunk 2 above)
Comment (mapk-amazon) [reply cli/src/pcluster/validators/efs_validators.py]: Replaced the wrong `name` by the `file-system-id`.

File: cli/tests/pcluster/validators/test_efs_validators.py (hunk 3 above)
Comment (mapk-amazon) [reply cli/tests/pcluster/validators/test_efs_validators.py]: Replaced the wrong `name` by the `file-system-id`.

**Review by dreambeyondorange — DISMISSED** [review]

**Comment by dreambeyondorange** [comment]
Approved but you need to rebase

**Comment by mapk-amazon** [comment]
Rebased to the latest develop. Needed to rewrite the history a bit... Should now be up-to-date.

**Review by dreambeyondorange — DISMISSED** [review]

**Review by dreambeyondorange — APPROVED** [review]

**Pull request merged**